File: server/DocumentsKM/Models/MarkLinkedDoc.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DocumentsKM.Models
{
    // Ссылочный документ марки
    public class MarkLinkedDoc
    {
        [Key]
        public int Id { get; set; }

        // Марка
        [Required]
        [ForeignKey("MarkId")]
        public virtual Mark Mark { get; set; }

        // Ссылочный документ
        [Required]
        [ForeignKey("LinkedDocId")]
        public virtual LinkedDoc LinkedDoc { get; set; }

        // Примечание
        [MaxLength(50)]
        public string Note { get; set; }
    }
}

File: client/src/model/MarkLinkedDoc.ts
import LinkedDoc from './LinkedDoc'

interface IMarkLinkedDoc {
    id: number
    linkedDoc: LinkedDoc
    note: string
}

class MarkLinkedDoc {
	id: number
    linkedDoc: LinkedDoc
    note: string

	constructor(obj?: IMarkLinkedDoc) {
		this.id = (obj && obj.id) || 0
        this.linkedDoc = (obj && obj.linkedDoc) || null
        this.note = (obj && obj.note) || ''
	}
}

export default MarkLinkedDoc
